integration-libs/opf/cta/components/opf-cta-scripts/opf-cta-scripts.service.ts

```diff
@@ -0,0 +1,186 @@
+        this.opfResourceLoaderService
+          .loadProviderResources(script.jsUrls, script.cssUrls)
+          .then(() => {
+            if (html) {
+              this.opfResourceLoaderService.executeScriptFromHtml(html);
+              resolve(script);
+            } else {
+              resolve(undefined);
+            }
+          })
+          .catch(() => {
+            resolve(undefined);
+          });
+      }
+    );
+  }
+
+  getCtaHtmlList(): Observable<OpfDynamicScript[]> {
+    let isDynamicCtaLocation = false;
+    return this.fillCtaScriptRequest().pipe(
+      concatMap((ctaScriptsRequest) => {
+        isDynamicCtaLocation =
+          !!ctaScriptsRequest?.scriptLocations?.length &&
+          !!ctaScriptsRequest?.scriptLocations.find((location) =>
+            DynamicCtaLocations.includes(location)
+          );
+        isDynamicCtaLocation &&
+          this.opfDynamicCtaService.registerScriptReadyEvent();
+        return this.fetchCtaScripts(ctaScriptsRequest);
+      }),
+      tap((scriptsResponse) => {
+        isDynamicCtaLocation &&
+          !!scriptsResponse.length &&
+          this.opfDynamicCtaService.initiateEvents();
+      }),
+      finalize(() => {
+        this.opfResourceLoaderService.clearAllProviderResources();
+        isDynamicCtaLocation && this.opfDynamicCtaService.stopEvents();
+      })
+    );
+  }
+
+  protected fetchCtaScripts(
+    ctaScriptsRequest: CtaScriptsRequest
+  ): Observable<OpfDynamicScript[]> {
+    return this.opfCtaFacade.getCtaScripts(ctaScriptsRequest).pipe(
+      concatMap((ctaScriptsResponse: CtaScriptsResponse) => {
+        if (!ctaScriptsResponse?.value?.length) {
+          return throwError(() => 'Invalid CTA Scripts Response');
+        }
+        const dynamicScripts = ctaScriptsResponse.value.map(
+          (ctaScript) => ctaScript.dynamicScript
+        );
+        return of(dynamicScripts);
+      }),
+      take(1)
+    );
+  }
+
+  protected fillCtaScriptRequest(): Observable<CtaScriptsRequest> {
+    let paymentAccountIds: number[];
+
+    return this.getPaymentAccountIds().pipe(
+      concatMap((accIds) => {
+        paymentAccountIds = accIds;
+        return this.getScriptLocation();
+      }),
+      concatMap((scriptsLocation: CtaScriptsLocation | undefined) => {
+        return this.fillRequestForTargetPage(
+          scriptsLocation,
+          paymentAccountIds
+        );
+      })
+    );
+  }
+
+  protected fillRequestForTargetPage(
+    scriptsLocation: CtaScriptsLocation | undefined,
+    paymentAccountIds: number[]
+  ): Observable<CtaScriptsRequest> {
+    if (!scriptsLocation) {
+      return throwError(() => 'Invalid Script Location');
+    }
+    const locationToFunctionMap: Record<
+      CtaScriptsLocation,
+      () => Observable<CtaScriptsRequest>
+    > = {
+      [CtaScriptsLocation.ORDER_HISTORY_PAYMENT_GUIDE]: () =>
+        this.opfStaticCtaService.fillCtaRequestforPagesWithOrder(
```

Comment:
[MEDIUM]
PascalCase typo `Requestfor` vs `RequestFor`

Note: analogical comment applies to many other methods in this PR